catalog/cataloger_diff.go

```diff
@@ -112,45 +171,130 @@ func (c *cataloger) getDiffSummary(ctx context.Context, tx db.Tx) (map[Differenc
+	// get child last commit of merge from parent
+	var childLastFromParentCommitID CommitID
+	query, args, err := sq.Select("MAX(commit_id) as max_child_commit").
 		From("catalog_commits").
 		Where("branch_id = ? AND merge_type = 'from_parent'", childID).
 		PlaceholderFormat(sq.Dollar).
 		ToSql()
 	if err != nil {
 		return fmt.Errorf("get child last commit sql: %w", err)
 	}
-	err = tx.Get(&maxChildMerge, maxChildQuery, args...)
+	err = tx.Get(&childLastFromParentCommitID, query, args...)
 	if err != nil {
 		return fmt.Errorf("get child last commit failed: %w", err)
 	}
-	diffResultsTableName, err := diffResultsTableNameFromContext(ctx)
+
+	diffResultsTableName, err := createDiffResultsTable(ctx, tx)
 	if err != nil {
 		return err
 	}
-	diffFromParentSQL, args, err := sqDiffFromParentV(parentID, childID, maxChildMerge, parentLineage, childLineage).
-		Prefix(`CREATE UNLOGGED TABLE ` + diffResultsTableName + " AS ").
-		PlaceholderFormat(sq.Dollar).
-		ToSql()
+
+	scannerOpts := DBScannerOptions{
+		After:            after,
+		AdditionalFields: []string{DBEntryFieldChecksum},
+	}
+	parentScanner := NewDBLineageScanner(tx, parentID, CommittedID, &scannerOpts)
+	childScanner := NewDBLineageScanner(tx, childID, UncommittedID, &scannerOpts)
+	childLineage, err := childScanner.ReadLineage()
 	if err != nil {
-		return fmt.Errorf("diff from parent sql: %w", err)
+		return err
 	}
-	if _, err := tx.Exec(diffFromParentSQL, args...); err != nil {
-		return fmt.Errorf("select diff from parent: %w", err)
+	batch := newDiffResultsBatchWriter(tx, diffResultsTableName)
+	var childEnt *DBScannerEntry
+	records := 0
+	for parentScanner.Next() {
+		// stop on limit
+		if limit > -1 && records >= limit {
+			break
+		}
+
+		// is parent element is relevant
+		parentEnt := parentScanner.Value()
+
+		// get next parent - next parentEnt that path >= child
+		childEnt, err = ScanDBEntryUntil(childScanner, childEnt, parentEnt.Path)
+		if err != nil {
+			return err
+		}
+
+		// point to matched child based on path
+		var matchedChild *DBScannerEntry
+		if childEnt != nil && childEnt.Path == parentEnt.Path {
+			matchedChild = childEnt
+		}
+		// diff between entries
+		diffType := evaluateFromParentElementDiffType(childID, childLastFromParentCommitID, childLineage, parentEnt, matchedChild)
+		if diffType == DifferenceTypeNone {
+			continue
+		}
+
+		diffRec := &diffResultRecord{
+			SourceBranch: parentID,
+			DiffType:     diffType,
+			Entry:        parentEnt.Entry,
+		}
+		if matchedChild != nil && matchedChild.BranchID == childID && diffType != DifferenceTypeConflict && diffType != DifferenceTypeRemoved {
+			diffRec.EntryCtid = &parentEnt.RowCtid
+		}
+		err = batch.Write(diffRec)
+		if err != nil {
+			return err
+		}
+		records++
 	}
-	return nil
+	if err := parentScanner.Err(); err != nil {
+		return err
+	}
+	return batch.Flush()
+}
+
+func lineageCommitIDByBranchID(lineage []lineageCommit, branchID int64) CommitID {
```

Comment:
Isn't there another very simmilar function?
Anyway - I think map is faster and simpler